core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/keymanager/imports/ImportKeystoresHandler.java

```diff
@@ -141,35 +137,63 @@ public void handle(final RoutingContext context) {
-      final String pubkey = pubkeysToImport.get(i);
-      try {
-        final String jsonKeystoreData = parsedBody.getKeystores().get(i);
-        final String password = parsedBody.getPasswords().get(i);
-
-        if (existingPubkeys.contains(pubkey)) {
-          // keystore already loaded
-          results.add(new ImportKeystoreResult(ImportKeystoreStatus.DUPLICATE, null));
-        } else {
-          validatorManager.addValidator(Bytes.fromHexString(pubkey), jsonKeystoreData, password);
-          results.add(new ImportKeystoreResult(ImportKeystoreStatus.IMPORTED, null));
-        }
-      } catch (final Exception e) {
-        // cleanup the current key being processed and continue
-        removeSignersAndCleanupImportedKeystoreFiles(List.of(pubkey));
-        results.add(
-            new ImportKeystoreResult(
-                ImportKeystoreStatus.ERROR, "Error importing keystore: " + e.getMessage()));
+      final String jsonKeystoreData = parsedBody.getKeystores().get(i);
+      final String password = parsedBody.getPasswords().get(i);
+      final String pubkey = getNormalizedPubKey(jsonKeystoreData);
+      if (existingPubkeys.contains(pubkey)) {
+        // keystore already loaded
+        duplicateData.add(
+            new ImportKeystoreData(
+                i, pubkey, jsonKeystoreData, password, new ImportKeystoreResult(DUPLICATE)));
+      } else {
+        toBeImportedData.add(
+            new ImportKeystoreData(
+                i, pubkey, jsonKeystoreData, password, new ImportKeystoreResult(IMPORTED)));
       }
     }
 
+    // process TO_BE_IMPORTED in parallel and add to validator manager. Change status to ERROR if
+    // any error happens.
+    final List<ImportKeystoreData> importedData =
+        toBeImportedData.parallelStream()
+            .peek(
+                data -> {
+                  try {
+                    validatorManager.addValidator(
+                        data.getPubKeyBytes(), data.getKeystoreJson(), data.getPassword());
+                  } catch (final Exception e) {
+                    data.setImportKeystoreResult(
+                        new ImportKeystoreResult(
+                            ImportKeystoreStatus.ERROR,
+                            "Error importing keystore: " + e.getMessage()));
+                  }
+                })
+            .toList();
+
+    // clean out any error results
+    removeSignersAndCleanupImportedKeystoreFiles(
+        importedData.stream()
+            .filter(
+                data -> data.getImportKeystoreResult().getStatus() == ImportKeystoreStatus.ERROR)
+            .map(ImportKeystoreData::getPubKey)
+            .toList());
+
+    // merge duplicate and imported data and sort
+    final List<ImportKeystoreResult> results =
+        Stream.concat(duplicateData.stream(), importedData.stream())
+            .sorted()
+            .map(ImportKeystoreData::getImportKeystoreResult)
+            .toList();
+
```

Comment:
This block of code in the ImportKeystoresHandler.handle is getting very long and hard to read can this be broken up?